test/regtest.py

```diff
@@ -104,43 +104,142 @@ def metadata(file):
       for verifier in meta['verifiers'][:100 if args.exhaustive else 1]:
         cmd += ['--verifier=' + verifier]
 
-        print "{0:>20} {1:>10}    :".format(memory, verifier),
+        str_result += "{0:>20} {1:>10}    :".format(memory, verifier)
 
         t0 = time.time()
         p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
         out, err  = p.communicate()
         elapsed = time.time() - t0
 
+        # get the test results
         result = get_result(out+err)
-
+        passed = timedout = unknown = False
         if result == meta['expect']:
-          print green('PASSED '),
-          passed += 1
+          str_result += 'PASSED '
+          passed = True
 
         elif result == 'timeout':
-          print red('TIMEOUT'),
-          timeouts += 1
+          str_result += 'TIMEOUT'
+          timedout = True
 
         elif result == 'unknown':
-          print red('UNKNOWN'),
-          unknowns += 1
+          str_result += 'UNKNOWN'
+          unknown = True
 
         else:
-          print red('FAILED '),
-          failed += 1
+          str_result += 'FAILED '
 
-        print '  [%.2fs]' % round(elapsed, 2)
-        sys.stdout.flush()
-  
-except KeyboardInterrupt:
-  pass
+        str_result += '  [%.2fs]' % round(elapsed, 2)
 
-print
-print ' PASSED count:', passed
-print ' FAILED count:', failed
+        # log the info for this test
+        logging.info(str_result)
```

Comment:
Looks like it’s OK, so long `multiprocessing.Pool` implements its pool with threads, not processes.

https://docs.python.org/2/howto/logging-cookbook.html#logging-to-a-single-file-from-multiple-processes